frontend/src/components/IssueCredential.vue

```diff
@@ -0,0 +1,236 @@
+            outlined
+            :disabled="this.$props.credDefId !== undefined"
+            dense
+            @change="credDefSelected"
+        ></v-select>
+      </v-card-text>
+      <v-card-text>
+        <h4 v-if="credDef && credDef.fields && credDef.fields.length">
+          Credential Content
+        </h4>
+        <v-row>
+          <v-col>
+            <v-text-field
+                v-for="field in credDef.fields"
+                :key="field.type"
+                :label="field.label"
+                placeholder
+                :rules="[(v) => !!v || 'Item is required']"
+                :required="field.required"
+                outlined
+                dense
+                @change="fieldChanged(field.type, $event)"
+            ></v-text-field>
+          </v-col>
+        </v-row>
+      </v-card-text>
+      <v-card-actions>
+        <v-layout align-end justify-end>
+          <v-btn color="secondary" text @click="cancel()">Cancel</v-btn>
+          <v-btn
+              :loading="this.isBusy"
+              color="primary"
+              text
+              @click="submit()"
+              :disabled="submitDisabled"
+          >Submit</v-btn
+          >
+        </v-layout>
+      </v-card-actions>
+    </v-card>
+  </v-container>
+</template>
+
+<script>
+  import { EventBus } from "@/main";
+  import { issuerService, partnerService } from "@/services";
+  import * as textUtils from "@/utils/textUtils";
+
+  export default {
+    name: "IssueCredential",
+    components: {},
+    props: {
+      partnerId: String,
+      credDefId: String,
+      partnerList: Array,
+      credDefList: Array
+    },
+    mounted() {
+      this.load();
+    },
+    data: () => {
+      return {
+        isLoading: false,
+        isBusy: false,
+        partners: [],
+        credDefs: [],
+        partner: {},
+        credDef: {},
+        credential: {},
+        credentialFields: {},
+        submitDisabled: true,
+      };
+    },
+    computed: {},
+    watch: {
+      partnerId(val) {
+        if (val) {
+         this.partner = this.partners.find(
+            (p) => p.value === val
+          );
+        }
+      },
+      credDefId(val) {
+        if (val) {
+          this.credDef = this.credDefs.find(
+            (p) => p.value === val
+          );
+          this.credDefSelected();
+        }
+      },
+      partnerList(val) {
+        this.partners = Array.from(val);
+      },
+      credDefList(val) {
+        this.credDefs = Array.from(val);
+      },
+    },
+    methods: {
+      async load() {
+        this.isLoading = true;
+        this.partners = [];
+        this.partner = {};
+        this.credDefs = [];
+        this.credDef = {};
+
+        // get partner list
+        if (!this.$props.partnerList || this.$props.partnerList.length === 0) {
+          const presp = await partnerService.listPartners();
+          if (presp.status === 200) {
+            this.partners = presp.data.map((p) => {
```

Comment:
I think you should filter for active connections or give the user at least a hint on the status of the partner connection.
Currently e.g. invitations would appear in this list.

Reply:
added an indicator, couldn't use the partner state indicator component directly but moved the logic into partnerUtils.